### PR TITLE
Support for DB2 and DB2400

### PR DESCRIPTION
**Pull request by marvindaviddiaz**

### CODE REVIEW

**Comment by bartoszwalacik** [comment]
hi, how can we test it? i mean how do I run db2 on local machine?

**Comment by marvindaviddiaz** [comment]
Hi! You can install the free version of DB2: DB2 Express-C
https://www-01.ibm.com/marketing/iwm/iwm/web/pick.do?source=swg-db2expressc&S_TACT=000000TA&lang=en_US

**Comment by bartoszwalacik** [comment]
are there docker images with db2 ?

**Comment by marvindaviddiaz** [comment]
Yes, but I have never tried it.
https://hub.docker.com/r/ibmcom/db2express-c/

**Comment by bartoszwalacik** [comment]
ok, do you need db2 dialect in Javers?

**Comment by marvindaviddiaz** [comment]
Really I already have Javers running on DB2 and DB2400 (db2 for AS400) and I wanted to share the classes.

**Comment by marvindaviddiaz** [comment]
The only problem I had: FK and PK cannot be a null (using db2). The following columns have constraints and are nullable.

jv_commit_property
 - property_name
 - property_value

jv_snapshot
 - global_id_fk
 - commit_fk

For that reason I had to create the schema manual, and Javers works fine.

**Comment by bartoszwalacik** [comment]
That's another issue but
cdo_snapshot.global_id_fk and cdo_snapshot.commit_fk can't be null. Maybe there is no db constraint for that, but for sure, JaVers won't insert into cdo_snapshot with null global_id_fk or commit_fk. 
property_name and property_value are not part of fk but an index

File: src/main/java/org/polyjdbc/core/dialect/DB2Dialect.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public String nextFromSequence(String sequenceName) {
+        return "values nextval for " + sequenceName ;
+    }
+
+    @Override
+    public DialectConstraints constraints() {
+        return constraints;
+    }
+
```
Comment (bartoszwalacik) [inline src/main/java/org/polyjdbc/core/dialect/DB2Dialect.java]: why not?
```
    @Override
    public DialectTypes types() {
        return new DB2DialectTypes();
    }
```
DB2DialectTypes seems not used

File: src/main/java/org/polyjdbc/core/dialect/DB2400Dialect.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @author Marvin Diaz
+ */
+public class DB2400Dialect extends DB2Dialect {
+
+    private DB2400DialectQueries queries = new DB2400DialectQueries();
+
+    @Override
+    public String getCode() {
+        return DialectRegistry.DB2.name();
```
Comment (bartoszwalacik) [inline src/main/java/org/polyjdbc/core/dialect/DB2400Dialect.java]: DB2400

**Comment by bartoszwalacik** [comment]
@marvindaviddiaz did you tested it :) ? I've pointed two bugs in this code ...

**Comment by marvindaviddiaz** [comment]
sorry man :) the errors are already corrected

**Pull request merged**